### PR TITLE
[reggen] Move prj metadata into main hjson 

### PR DESCRIPTION
**Pull request by msfschaffner**
This is in preparation for #15333 (version registers). Moving the meta data into the main IP hjson will facilitate creating the version register record, since everything comes from one source file. Otherwise we would have to discover and plumb the `prj.hjson` files through reggen / topgen as well.

### CODE REVIEW

**Comment by tjaychen** [comment]
sorry a bit behind on reviews, i'll have a look tonight!

File: hw/top_earlgrey/doc/design/index.md [hunk 1]
```diff
@@ -146,19 +146,15 @@ See that specification for generic details on this scheme.
 
 #### Chip IO Peripherals
 
-##### Pin Multiplexor and Pad Control Module (`pinmux`)
+##### Pin Multiplexor Module (`pinmux`)
 
-**TODO: this section needs to be updated to reflect the pinmux/padctrl merger**
-
-The pin multiplexor's purpose is to route between peripherals and the available multiplexable IO (`MIO_00 .. MIO_31`) of the chip.
+The pin multiplexor's purpose is to route between peripherals and the available multiplexable IO of the chip.
```
Comment (sriyerg) [inline hw/top_earlgrey/doc/design/index.md:151]: s/multiplexor/multiplexer? 
https://en.wikipedia.org/wiki/Multiplexer

Comment (sriyerg) [reply hw/top_earlgrey/doc/design/index.md:151]: Ha! I just realized the wiki does mention multiplexor as an alternate spelling....

**Review by sriyerg — APPROVED** [review]
LGTM, Thanks!

**Comment by tjaychen** [comment]
lgtm! just a quick question, i saw a github thing that said codeowners file contains an error, any idea what that is?

**Comment by msfschaffner** [comment]
> lgtm! just a quick question, i saw a github thing that said codeowners file contains an error, any idea what that is?

good point, I just saw that, too! it looks like these were pre-existing and not related to this change:
```
Unknown owner on line 33: make sure @Jacob-Levy exists and has write access to the repository
util/regtool.py     @Jacob-Levy
Unknown owner on line 34: make sure @Jacob-Levy exists and has write access to the repository
util/reggen/        @Jacob-Levy
Unknown owner on line 37: make sure @Jacob-Levy exists and has write access to the repository
…      @msfschaffner @tjaychen @Jacob-Levy
Unknown owner on line 38: make sure @Jacob-Levy exists and has write access to the repository
…      @msfschaffner @tjaychen @Jacob-Levy
Unknown owner on line 54: make sure @Jacob-Levy exists and has write access to the repository
/hw/top_*/ip/ast        @Jacob-Levy
Unknown owner on line 72: make sure @milesdai exists and has write access to the repository
…i/                @drewmacrae @milesdai @rswarbrick
Unknown owner on line 74: make sure @milesdai exists and has write access to the repository
…ure-pipelines.yml @drewmacrae @milesdai @rswarbrick
```

looks like it GH spits out warnings if a handle is not on the committer list.

**Comment by msfschaffner** [comment]
@arnonsha @Jacob-Levy you may have to add the missing fields to your internal AST hjson file so that regtool does not throw any errors.

**Comment by meisnere** [comment]
Nothing need to be changed on our side as we don't see any errors.

**Pull request merged**